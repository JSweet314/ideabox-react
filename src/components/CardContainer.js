import React from 'react';
import IdeaCard from './IdeaCard.js';
import QualityFilter from './QualityFilter.js';
import PropTypes from 'prop-types';
import '../styles/CardContainer.css';

const CardContainer = (
  { visibleIdeas, updateIdeaQuality, removeIdea, searchIdeas, filterByQuality }
) => {
  return (
    <div className="card-container">
      <input
        className="search"
        type="search"
        placeholder="SEARCH"
        onChange={(e) => searchIdeas(e.target.value)}
      />
      <QualityFilter filterByQuality={filterByQuality} />
      {
        visibleIdeas.map((idea, index) =>
          <IdeaCard 
            {...idea}
            key={index}
            updateIdeaQuality={updateIdeaQuality}
            removeIdea={removeIdea}
          />
        )
      }
    </div>
  );
};

CardContainer.defaultProps = {
  visibleIdeas: []
};

CardContainer.propTypes = {
  visibleIdeas: PropTypes.arrayOf(PropTypes.object),
  removeIdea: PropTypes.func.isRequired,
  updateIdeaQuality: PropTypes.func.isRequired,
  searchIdeas: PropTypes.func.isRequired,
  filterByQuality: PropTypes.func.isRequired
};

export default CardContainer;