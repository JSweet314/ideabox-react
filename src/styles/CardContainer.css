.card-container {
  width: 100%;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

.search {
  font-family: 'OpenSans', sans-serif;
  margin: 1rem 0 0 0;
  height: 2rem;
  padding: 0 0 0 1rem;
  width: 51%;
  border-style: unset;
  border: 2px solid lightgray;
  font-size: 1rem;
}

.search::placeholder {
  font-family: 'Open Sans', sans-serif;
}