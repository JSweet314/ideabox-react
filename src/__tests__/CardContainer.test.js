import React from 'react';
import {shallow, mount} from 'enzyme';
import CardContainer from '../scripts/CardContainer.js';

describe('CardContainer', () => {
  let cardContainer, idea1, idea2, visibleIdeas;

  beforeEach(() => {
    idea1 = {
      id: 1234,
      title: 'Hello',
      body: 'World!',
      quality: 'plausible'
    };

    idea2 = {
      id: 4321,
      title: 'New App!',
      body: 'Woo Hoo',
      quality: 'genius'
    };

    visibleIdeas = [idea1, idea2];
    
    cardContainer = shallow(
      <CardContainer
        visibleIdeas={visibleIdeas}
        updateIdeaQuality={jest.fn()}
        removeIdea={jest.fn()}
        searchIdeas={jest.fn()}
      />
    );
  });

  it('should render our good friend, CardContainer', () => {
    expect(cardContainer).toMatchSnapshot();
    
    cardContainer = shallow(
      <CardContainer
        visibleIdeas={[]}
        updateIdeaQuality={jest.fn()}
        removeIdea={jest.fn()}
        searchIdeas={jest.fn()}
      />
    );

    expect(cardContainer).toMatchSnapshot();
  });

  it('should have a prop visibleIdeas as well as methods from the app component', () => {
    expect(cardContainer.props('visibleIdeas')).toBeDefined();
    expect(cardContainer.props('updateIdeaQuality')).toBeDefined();
    expect(cardContainer.props('removeIdea')).toBeDefined();
    expect(cardContainer.props('searchIdeas')).toBeDefined();
  });
});
