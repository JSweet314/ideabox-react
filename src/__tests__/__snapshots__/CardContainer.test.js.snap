// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer should render without crashing 1`] = `
<div
  className="card-container"
>
  <input
    className="search"
    onChange={[Function]}
    placeholder="SEARCH"
    type="search"
  />
  <IdeaCard
    body="World!"
    id={1234}
    key="0"
    quality="plausible"
    removeIdea={[Function]}
    title="Hello"
    updateIdeaQuality={[Function]}
  />
  <IdeaCard
    body="Woo Hoo"
    id={4321}
    key="1"
    quality="genius"
    removeIdea={[Function]}
    title="New App!"
    updateIdeaQuality={[Function]}
  />
</div>
`;

exports[`CardContainer should render without crashing 2`] = `
<div
  className="card-container"
>
  <input
    className="search"
    onChange={[Function]}
    placeholder="SEARCH"
    type="search"
  />
</div>
`;
